File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/verify/ResetPasswordService.java
package uk.gov.justice.digital.hmpps.oauth2server.verify;

import uk.gov.justice.digital.hmpps.oauth2server.verify.ResetPasswordServiceImpl.NotificationClientRuntimeException;
import uk.gov.justice.digital.hmpps.oauth2server.verify.ResetPasswordServiceImpl.ResetPasswordException;

import java.util.Optional;

public interface ResetPasswordService extends PasswordService {
    Optional<String> requestResetPassword(String inputUsername, String url) throws NotificationClientRuntimeException;

    void setPassword(String token, String password);

    String moveTokenToAccount(String token, String username) throws ResetPasswordException;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/ResetPasswordService.kt
package uk.gov.justice.digital.hmpps.oauth2server.verify

import uk.gov.justice.digital.hmpps.oauth2server.verify.ResetPasswordServiceImpl.NotificationClientRuntimeException
import uk.gov.justice.digital.hmpps.oauth2server.verify.ResetPasswordServiceImpl.ResetPasswordException
import java.util.Optional

interface ResetPasswordService : PasswordService {
  @Throws(NotificationClientRuntimeException::class)
  fun requestResetPassword(usernameOrEmailAddress: String, url: String): Optional<String>
  override fun setPassword(token: String, password: String?)

  @Throws(ResetPasswordException::class)
  fun moveTokenToAccount(token: String, username: String?): String
}
